src/main/java/com/cx/restclient/ast/AstClient.java

```diff
@@ -187,4 +207,55 @@ protected void handleInitError(IOException e) {
         String message = String.format("Failed to init %s client.", getScannerDisplayName());
         throw new CxClientException(message, e);
     }
+
+    protected HttpResponse submitAllSourcesFromLocalDir(String projectId, String zipFilePath) throws IOException {
+        log.info("Using local directory flow.");
+
+        PathFilter filter = new PathFilter(config.getOsaFolderExclusions(), config.getOsaFilterPattern(), log);
+        String sourceDir = config.getEffectiveSourceDirForDependencyScan();
```

Comment:
This logic is SCA-specific. The getOsaFolderExclusions, getOsaFilterPattern, getEffectiveSourceDirForDependencyScan properties should only be used in AstScaClient. AstSastClient should use different properties.